crates/biome_js_formatter/src/js/expressions/call_arguments.rs

```diff
@@ -1067,8 +1067,11 @@ fn can_group_expression_argument(
                     .map_or(true, |any_type| match any_type {
                         AnyTsReturnType::AnyTsType(AnyTsType::TsReferenceType(_)) => match &body {
                             AnyJsFunctionBody::JsFunctionBody(body) => {
-                                body.statements().iter().any(|statement| {
-                                    !matches!(statement, AnyJsStatement::JsEmptyStatement(_))
+                                body.statements().iter().any(|statement| match statement {
+                                    AnyJsStatement::JsEmptyStatement(s) => {
+                                        comments.has_comments(s.syntax())
+                                    }
```

Comment:
Could you add a short comment here explaining why this check is necessary? Since this is such a deeply-nested match, it can be hard to know which parts are doing what, and some words will help clear it up whenever people encounter this in the future. Something like:

> When the body contains an empty statement, comments in the body will get attached to that statement rather than the body itself, so they need to be checked for comments as well to ensure that the body is still considered groupable when those empty statements are removed by the printer.